https://bugs.gentoo.org/show_bug.cgi?id=231962

sci-physics/geant-4.9.1_p02 wants to be able to get an openinventor with
USE=openinventor.  However, on sparc we must package.use.mask this option
because openinventor will not build.  It will not build because its
architecture file (machine.h) does not know about sparc linux (or any
big-endian linux for that matter).  With the following little patch,
openinventor will build and install on sparc, and the demos all seem good.  I
do not know if everything is correct, however, because this change only matters
in a couple classes (SoInput, SoOutput), and I don't know if the demos hit
them.  No attachment because the test patch changes only one line.
===============================================
polylepis inventor # diff -u lib/database/include/machine.h-
lib/database/include/machine.h
--- lib/database/include/machine.h-     2008-07-16 12:54:21.000000000 +0000
+++ lib/database/include/machine.h      2008-07-16 12:54:34.000000000 +0000
@@ -106,7 +106,7 @@
  * SUN/BSD machine dependent setup
  */

-#if sun
+#if sun || sparc

 #define MACHINE_WORD_FORMAT    DGL_BIG_ENDIAN
 #define MACHINE_FLOAT_FORMAT   DGL_BIG_IEEE
================================================

